skimage/measure/tests/test_regionprops.py

```diff
@@ -1236,6 +1236,16 @@ def test_cache():
     assert np.any(f0 != f1)
 
 
+def test_cache_memory():
+    SAMPLE_mod = SAMPLE.copy()
+    region = regionprops(SAMPLE_mod, cache=False)[0]
+    # Acces one property
+    _ = region.image_filled
+
+    # Cache should be empty
+    assert not region._cache
```

Comment:
A bit more expressive:

```suggestion
    assert region._cache == dict()
```